### PR TITLE
SP-49, SP-50 and SP-51: Admin dashboard fixes for Category & its descriptions

### PR DESCRIPTION
**Pull request by Prasad1102**
SP-49 : Admin Dashboard - Add Filter on Categories Page
SP-50 : Admin Dashboard - Cached Data in Descriptions Page
SP-51 : Admin Dashboard - Video Error

### CODE REVIEW

File: src/components/data_management/Categories.jsx [hunk 1]
```diff
@@ -95,6 +103,30 @@ const Categories = () => {
...
+    const selectedFilter = event.target.value;
+    setFilter(selectedFilter);
+
+    if (selectedFilter === "all") {
+      setState((prevState) => ({
+        ...prevState,
+        filteredCategories: prevState.categories,
+      }));
+    } else {
+      const isActive = selectedFilter === "active";
```
Comment (GaneshNavale) [inline src/components/data_management/Categories.jsx:119]: Use Server Side filtering

File: src/components/data_management/DescriptionModal.jsx [hunk 2]
```diff
@@ -241,6 +254,7 @@ const DescriptionModal = (props) => {
           )}
 
           {description.media_type === "video" && (
+            // sx={{ marginBottom: 4 }} will add in grid
```
Comment (GaneshNavale) [inline src/components/data_management/DescriptionModal.jsx:257]: remove

File: src/components/data_management/DescriptionModal.jsx [hunk 3]
```diff
@@ -144,7 +144,13 @@ const DescriptionModal = (props) => {
         formData.append("description[images][]", image);
       });
     } else if (description.media_type === "video" && description.video) {
-      formData.append("description[video]", description.video);
+      // formData.append("description[video]", description.video);
+      // Prasad Shelke Change added if condition
+      if (description.video instanceof File) {
+        formData.append("description[video]", description.video);
+      } else {
+        console.error("Video is not a file object:", description.video);
```
Comment (GaneshNavale) [inline src/components/data_management/DescriptionModal.jsx:152]: Check if we can utilize onChange event on media_type to reset video & images

**Review by GaneshNavale — CHANGES_REQUESTED** [review]
Please address the suggested changes.

File: src/utils/VideoUploader.js [hunk 4]
```diff
@@ -10,7 +10,11 @@ import {
 import DeleteIcon from "@mui/icons-material/Delete";
 
 const VideoUploader = ({ videoUrl, onVideoChange }) => {
+  console.log("videoUrl", videoUrl);
   const [video, setVideo] = useState(videoUrl || null);
+  useEffect(() => {
+    console.log("video", video);
+  }, [video]);
```
Comment (GaneshNavale) [inline src/utils/VideoUploader.js:17]: Remove

File: src/utils/VideoUploader.js [hunk 5]
```diff
@@ -21,11 +21,10 @@ const VideoUploader = ({ videoUrl, onVideoChange }) => {
         setAlert({ open: true, message: "Please upload a valid MP4 video." });
         return;
       }
-      setLoading(true);
       const videoURL = URL.createObjectURL(file);
       setVideo(videoURL);
+      console.log("Video File:", file); // Add this line to verify the file type
       onVideoChange(file); // Send the file back to the parent
-      setLoading(false);
```
[the remark this answers is not in the document]
Comment (Prasad1102) [reply src/utils/VideoUploader.js:28]: Changes Completed

File: src/components/data_management/DescriptionModal.jsx (hunk 2 above)
Comment (Prasad1102) [reply src/components/data_management/DescriptionModal.jsx]: Changes are resolved

File: src/utils/VideoUploader.js (hunk 4 above)
Comment (Prasad1102) [reply src/utils/VideoUploader.js:17]: changes are completed

**Pull request merged**